pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1, date(2008, 10, 23)],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1, date(2008, 10, 23)],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, date(2008, 10, 23)],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, date(2008, 10, 23)],
+    ], columns=['lat', 'lon', 'datetime', 'id', 'date'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert DATE not in move_df
+
+    move_df.generate_date_features()
+    assert_frame_equal(move_df, expected)
+
+
+def test_generate_hour_features():
+    move_df = _default_move_df()
+
+    new_move_df = move_df.generate_hour_features(inplace=False)
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1, 5],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1, 5],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 5],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 5],
+    ], columns=['lat', 'lon', 'datetime', 'id', 'hour'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert HOUR not in move_df
+
+    move_df.generate_hour_features()
+    assert_frame_equal(move_df, expected)
+
+
+def test_generate_day_of_the_week_features():
+    move_df = _default_move_df()
+
+    new_move_df = move_df.generate_day_of_the_week_features(inplace=False)
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1, 'Thursday'],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1, 'Thursday'],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 'Thursday'],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 'Thursday'],
+    ], columns=['lat', 'lon', 'datetime', 'id', 'day'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert DAY not in move_df
+
+    move_df.generate_day_of_the_week_features()
+    assert_frame_equal(move_df, expected)
+
+
+def test_generate_weekend_features():
+    move_df = _default_move_df()
+
+    new_move_df = move_df.generate_weekend_features(inplace=False)
+    expected = DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1, 0],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1, 0],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 0],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2, 0],
+    ], columns=['lat', 'lon', 'datetime', 'id', 'weekend'], index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(new_move_df, expected)
+    assert isinstance(new_move_df, PandasMoveDataFrame)
+    assert WEEK_END not in move_df
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574862)